src/PromiseType.cc

```diff
@@ -37,7 +33,7 @@ static bool onResolvedCb(JSContext *cx, unsigned argc, JS::Value *vp) {
   // Convert the Promise's result (either fulfilled resolution or rejection reason) to a Python object
   //  The result might be another JS function, so we must keep them alive
   JS::RootedValue resultArg(cx, args[0]);
-  PyObject *result = pyTypeFactory(cx, resultArg)->getPyObject();
+  PyObject *result = pyTypeFactory(cx, resultArg);
```

Comment:
`result` needs to be `Py_DECREF`'d before this function call ends regardless of which path we take (currently its only `DECREF`'d if the promise is rejected (see line 45).

Reply:
done